temporal-test-server/src/main/java/io/temporal/internal/testservice/TestWorkflowMutableStateImpl.java

```diff
@@ -250,6 +250,25 @@ private StartWorkflowExecutionRequest.Builder validateStartWorkflowExecutionRequ
     if (request.hasRetryPolicy()) {
       request.setRetryPolicy(validateAndOverrideRetryPolicy(request.getRetryPolicy()));
     }
+    if (request.getLinksCount() > 0) {
+      if (request.getLinksCount() > 10) {
+        throw Status.INVALID_ARGUMENT
+            .withDescription(
+                String.format(
+                    "cannot attach more than %d links per request, got %d",
+                    10, request.getLinksCount()))
+            .asRuntimeException();
+      }
+      for (Link l : request.getLinksList()) {
+        if (l.getSerializedSize() > 4000) {
```

Comment:
Note that I improved the validation here: https://github.com/temporalio/temporal/pull/6629/files#diff-c9a59445e489bd70346630f2773ad69e1723b2298fadbae50ff5c1a424917e1cR4613.
Might as well copy that logic while we're at it.